### PR TITLE
DDCE-683: Added summary page for section 4 : Charitable objectives

### PR DESCRIPTION
**Pull request by robert-webb-hmrc**
# DDCE-683

##New feature

DDCE-683: Summary page for Section 4: Charity Objectives

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/controllers/operationsAndFunds/CharityObjectivesSummaryController.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import pages.sections.Section4Page
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.UserAnswerRepository
+import viewmodels.operationsAndFunds.CharityObjectivesSummaryHelper
+import views.html.CheckYourAnswersView
+
+import scala.concurrent.Future
+
+class CharityObjectivesSummaryController @Inject()(
+                                                    val sessionRepository: UserAnswerRepository,
```
Comment (anburoach) [inline app/controllers/operationsAndFunds/CharityObjectivesSummaryController.scala:35]: Alignment: Please move injected objects to left side... refer other controllers

File: app/navigation/ObjectivesNavigator.scala [hunk 2]
```diff
@@ -49,6 +49,21 @@ class ObjectivesNavigator @Inject()(implicit frontendAppConfig: FrontendAppConfi
 
   private val checkRouteMap: Page => UserAnswers => Call = {
 
+    case CharitableObjectivesPage => userAnswers: UserAnswers => userAnswers.get(CharitableObjectivesPage) match {
```
Comment (anburoach) [inline app/navigation/ObjectivesNavigator.scala:52]: CheckMode changes are affecting coverage so please remove check mode changes or add corresponding test.
Note: Check mode changes are out of scope for this ticket

File: test/controllers/operationsAndFunds/CharityObjectivesSummaryControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      )
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    reset(mockUserAnswerRepository)
+  }
+
+  val controller: CharityObjectivesSummaryController = inject[CharityObjectivesSummaryController]
+
+  "Regulators Controller" must {
```
Comment (anburoach) [inline test/controllers/operationsAndFunds/CharityObjectivesSummaryControllerSpec.scala:52]: Please change Regulators to CharityObjectivesSummary

File: test/pages/operationsAndFunds/CharityObjectivesSummaryPageSpec.scala [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * limitations under the License.
+ */
+
+package pages.operationsAndFunds
+
+import pages.behaviours.PageBehaviours
+
+class CharityObjectivesSummaryPageSpec extends PageBehaviours {
+
+  "NIRegulatorRegNumberPage" must {
```
Comment (anburoach) [inline test/pages/operationsAndFunds/CharityObjectivesSummaryPageSpec.scala:23]: Please change NIRegulatorRegNumberPage to CharityObjectivesSummaryPage

File: test/viewModels/operationsAndFunds/CharityObjectivesSummaryHelperSpec.scala [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        )
+      }
+    }
+
+    "For the Charitable Purposes answer" must {
+
+      "have a correctly formatted summary list row" in {
+
+        val purposeList =
+          s"""<div>${messages(s"charitablePurposes.$other")}</div><div>${messages(s"charitablePurposes.$armedForcesOfTheCrown")}</div><div>${messages(s"charitablePurposes.$animalWelfare")}</div><div>${messages(s"charitablePurposes.$religion")}</div><div>${messages(s"charitablePurposes.$humanRights")}</div><div>${messages(s"charitablePurposes.$healthOrSavingOfLives")}</div><div>${messages(s"charitablePurposes.$citizenshipOrCommunityDevelopment")}</div><div>${messages(s"charitablePurposes.$environmentalProtection")}</div><div>${messages(s"charitablePurposes.$reliefOfPoverty")}</div><div>${messages(s"charitablePurposes.$artsCultureOrScience")}</div><div>${messages(s"charitablePurposes.$education")}</div><div>${messages(s"charitablePurposes.$reliefOfYouthAge")}</div><div>${messages(s"charitablePurposes.$amateurSport")}</div>""".stripMargin
```
Comment (anburoach) [inline test/viewModels/operationsAndFunds/CharityObjectivesSummaryHelperSpec.scala:73]: This line has more than 800 characters.... could u plz split this?

File: app/navigation/ObjectivesNavigator.scala (hunk 2 above)
Comment (robert-webb-hmrc) [reply app/navigation/ObjectivesNavigator.scala:52]: Tests added
